pkg/api/router.go

```diff
@@ -66,16 +66,13 @@ func GenerateRouter(apiConf *Config, client *pgclient.Client) (*mux.Router, erro
 	router.Path("/delete_series").Methods(http.MethodPut, http.MethodPost).HandlerFunc(deleteHandler)
 
 	queryable := client.Queryable()
-	queryEngine, err := query.NewEngine(log.GetLogger(), apiConf.MaxQueryTimeout, apiConf.LookBackDelta, apiConf.SubQueryStepInterval, apiConf.MaxSamples, apiConf.EnabledFeatureMap)
-	if err != nil {
-		return nil, fmt.Errorf("creating query-engine: %w", err)
-	}
+	queryEngine := client.QueryEngine()
 
 	apiV1 := router.PathPrefix("/api/v1").Subrouter()
 	queryHandler := timeHandler(metrics.HTTPRequestDuration, "query", Query(apiConf, queryEngine, queryable, updateQueryMetrics))
 	apiV1.Path("/query").Methods(http.MethodGet, http.MethodPost).HandlerFunc(queryHandler)
 
-	queryRangeHandler := timeHandler(metrics.HTTPRequestDuration, "query_range", QueryRange(apiConf, queryEngine, queryable, updateQueryMetrics))
+	queryRangeHandler := timeHandler(metrics.HTTPRequestDuration, "query_range", QueryRange(apiConf, promqlConf, queryEngine, queryable, updateQueryMetrics))
```

Comment:
Nit: Seems like queryEngine is used in only one place so why just not inline it in function call

Reply:
Its used in line 72 too.